### PR TITLE
Integrate TRS

### PR DESCRIPTION
**Pull request by shgandhi**
This PR adds TRS and its unit tests to concord-bft. Please refer to individual commits for more information.

### CODE REVIEW

File: kvbc/test/CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,20 @@ if(Boost_FOUND)
...
+        kvbc_app_filter/kvbc_filter_test.cpp)
+
+target_link_libraries(kvbc_filter_test
+        ${Boost_LIBRARIES}
+        GTest::Main
+        GTest::GTest
+        ${yaml-cpp_LIBRARIES}
+        kvbc
+        value-with-trids-proto
+		concordbft_reconfiguration
```
Comment (robem) [inline kvbc/test/CMakeLists.txt:26]: tabs/spaces off

File: thin-replica-server/test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        concord_trs
+        logging)
+
+add_executable(trs_sub_buffer_test
+        trs_sub_buffer_test.cpp)
+target_include_directories(trs_sub_buffer_test PUBLIC kvbc/include)
+target_link_libraries(trs_sub_buffer_test
+        ${Boost_LIBRARIES}
+        GTest::Main
+	GTest::GTest
```
Comment (robem) [inline thin-replica-server/test/CMakeLists.txt:24]: indentation, same as above.

**Review by robem — APPROVED** [review]
Lgtm, thanks!

**Comment by shgandhi** [comment]
Thanks for the review @robem, updated the PR according to your comments.

**Pull request merged**